src/sbt/probably.scala

```diff
@@ -0,0 +1,120 @@
+/*
+
+    Probably, version 0.3.0. Copyright 2017-20 Jon Pretty, Propensive OÜ.
+
+    The primary distribution site is: https://propensive.com/
+
+    Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+    compliance with the License. You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software distributed under the License is
+    distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and limitations under the License.
+
+*/
+package probably.sbt
+
+import sbt.testing
+import probably._
+
+class Framework extends testing.Framework {
+
+  override def name(): String = "Probably"
+
+  override def fingerprints(): Array[testing.Fingerprint] =
+    Array(new testing.SubclassFingerprint {
+      def isModule: Boolean = true
+      def superclassName(): String = "probably.Suite"
+      def requireNoArgConstructor(): Boolean = true
+    })
+
+  override def runner(_args: Array[String],
+                      _remoteArgs: Array[String],
+                      _testClassLoader: ClassLoader): testing.Runner =
+    new testing.Runner {
+      def args(): Array[String] = _args
+      def remoteArgs(): Array[String] = _remoteArgs
+      def done(): String = ""
+      def tasks(list: Array[testing.TaskDef]): Array[testing.Task] =
+        list.map(t => Task(t, _args))
+      def receiveMessage(msg: String): Option[String] = None  
+    }
+}
+
+case class Task(
+    taskDef: testing.TaskDef, 
+    args: Array[String]) extends testing.Task {
+
+  val suite: Suite =
+    Class
+      .forName(taskDef.fullyQualifiedName() + "$")
+      .getDeclaredField("MODULE$")
+      .get(null)
+      .asInstanceOf[Suite]    
+  
+  override def execute(
+      eh: testing.EventHandler, 
+      ls: Array[testing.Logger]): Array[testing.Task] = {
+
+    def log(l: testing.Logger => Unit): Unit = ls.foreach(l) 
+
+    val test = new Runner(args.map(Runner.TestId).to[Set])
+    suite.run(test)
+    test.report().results.foreach(
+      s =>
+        if (s.outcome.passed) {
+          log(passed(s.indent * 2, s.name))
+          eh.handle(Success(taskDef, s.ttot))
+        } else if (s.outcome.failed) {
+          val debug =
+            if (s.outcome.debug.isEmpty) "" else s" (${s.outcome.debug})"
+          log(passed(s.indent * 2, s"${s.name}$debug"))
+          eh.handle(Failure(taskDef, s.ttot))
+      }
+    )
```

Comment:
Slight reformatting:
```suggestion
    test.report().results.foreach { s =>
      if (s.outcome.passed) {
        log(passed(s.indent*2, s.name))
        eh.handle(Success(taskDef, s.ttot))
      } else if (s.outcome.failed) {
        val debug = s.outcome.debug.fold("") { d => s" (${d})" }
        log(passed(s.indent*2, s"${s.name}$debug"))
        eh.handle(Failure(taskDef, s.ttot))
      }
    }
```